src/main/java/net/datafaker/providers/healthcare/Disease.java

```diff
@@ -0,0 +1,113 @@
+package net.datafaker.providers.healthcare;
+
+import net.datafaker.providers.base.AbstractProvider;
+
+import java.util.List;
+import java.util.function.Supplier;
+
+/**
+ * Generate random, different kinds of disease.
+ *
+ * @since 0.8.0
+ */
+public class Disease extends AbstractProvider<HealthcareProviders> {
+
+    static final String INTERNAL_DISEASE_KEY = "healthcare.disease.internal_disease";
+    static final String NEUROLOGICAL_DISEASE_KEY = "healthcare.disease.neurology";
+    static final String SURGICAL_DISEASE_KEY = "healthcare.disease.surgery";
+    static final String PAEDIATRIC_DISEASE_KEY = "healthcare.disease.paediatrics";
+    static final String GYNECOLOGY_AND_OBSTETRICS_DISEASE_KEY = "healthcare.disease.gynecology_and_obstetrics";
+    static final String OPHTHALMOLOGY_AND_OTORHINOLARYNGOLOGY_DISEASE_KEY = "healthcare.disease.ophthalmology_and_otorhinolaryngology";
+    static final String DERMATOLOGY_DISEASE_KEY = "healthcare.disease.dermatology";
+    
+    /**
+     * Create a constructor for Disease
+     *
+     * @param faker The Faker instance for generating random, different kinds of disease, e.g. the internal disease.
+     */
+    protected Disease(HealthcareProviders faker) {
+        super(faker);
+    }
+
+    public String icd10() {
+        return faker.regexify(resolve("healthcare.disease.icd10"));
+    }
+
+    public String anyDisease() {
+        List<Supplier<String>> providers = List.of(
+            this::internalDisease,
+            this::neurology,
+            this::surgery,
+            this::paediatrics,
+            this::gynecologyAndObstetrics,
+            this::ophthalmologyAndOtorhinolaryngology,
+            this::dermatology
+        );
```

Comment:
I don't see a reason to create this list on each method call, it could be extracted in a const imho

Reply:
Done